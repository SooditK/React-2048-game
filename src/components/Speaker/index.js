import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import speakerOn from '../../assets/svg/speaker-on.svg';
import speakerOff from '../../assets/svg/speaker-off.svg';

// Speaker, keep self state for switch button icon
export default class Speaker extends Component {
  static propTypes = {
    onClick: PropTypes.func.isRequired
  };

  constructor(...args) {
    super(...args);

    this.state = {
      speaker: speakerOn
    };
  }

  handleClick = () => {
    const { speaker } = this.state;
    const prevStatus = speaker === speakerOn;
    this.setState({
      speaker: prevStatus ? speakerOff : speakerOn
    });
    this.props.onClick(!prevStatus);
  };

  render() {
    const { speaker } = this.state;
    return (
      <Button onClick={this.handleClick}>
        <img src={speaker} alt="speaker" />
      </Button>
    );
  }
}
