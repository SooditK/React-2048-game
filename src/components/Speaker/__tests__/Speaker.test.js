import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Speaker from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Speaker />', () => {
  it('component render', () => {
    const speaker = renderer.create(<Speaker onClick={() => {}} />).toJSON();
    expect(speaker).toMatchSnapshot();
  });

  it('click event', () => {
    const spy = jest.fn();
    const speaker = mount(<Speaker onClick={spy} />);
    speaker.find('[alt="speaker"]').simulate('click');
    expect(spy.mock.calls.length).toBe(1);
    speaker.find('[alt="speaker"]').simulate('click');
    expect(spy.mock.calls.length).toBe(2);
  });
});
