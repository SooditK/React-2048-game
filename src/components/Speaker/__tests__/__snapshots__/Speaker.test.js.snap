// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Speaker /> component render 1`] = `
<button
  className="btn-flat btn-md btn-default"
  onClick={[Function]}
>
  <img
    alt="speaker"
    src="speaker-on.svg"
  />
</button>
`;
